test/plog/LogStreamTest.cpp

```diff
@@ -0,0 +1,183 @@
+            .then([this] { return runTest2(); })
+            .then([this] { return runTest3(); })
+            .then([this] { return runTest4(); })
+            .then([this] {
+                K2INFO("======= All tests passed ========");
+                exitcode = 0;
+            })
+            .handle_exception([this](auto exc) {
+                try {
+                    std::rethrow_exception(exc);
+                } catch (RPCDispatcher::RequestTimeoutException& exc) {
+                    K2ERROR("======= Test failed due to timeout ========");
+                    exitcode = -1;
+                } catch (std::exception& e) {
+                    K2ERROR("======= Test failed with exception [" << e.what() << "] ========");
+                    exitcode = -1;
+                }
+            })
+            .finally([this] {
+                K2INFO("======= Test ended ========");
+                seastar::engine().exit(exitcode);
+            });
+        });
+        _testTimer.arm(0ms);
+        return seastar::make_ready_future<>();
+    }
+
+
+    seastar::future<> runTest1() {
+        K2INFO(">>> Test1: create a persistence cluster");
+        dto::PersistenceGroup group1{.name="Group1", .plogServerEndpoints = _plogConfigEps()};
+        dto::PersistenceCluster cluster1;
+        cluster1.name="Persistence_Cluster_1";
+        cluster1.persistenceGroupVector.push_back(group1);
+
+        auto request = dto::PersistenceClusterCreateRequest{.cluster=std::move(cluster1)};
+        return RPC()
+        .callRPC<dto::PersistenceClusterCreateRequest, dto::PersistenceClusterCreateResponse>(dto::Verbs::CPO_PERSISTENCE_CLUSTER_CREATE, request, *_cpoEndpoint, 1s)
+        .then([](auto&& response) {
+            auto& [status, resp] = response;
+            K2EXPECT(status, Statuses::S201_Created);
+        });
+    }
+
+    seastar::future<> runTest2() {
+        K2INFO(">>> Test2: init and create a logstream");
+        ConfigVar<String> configEp("cpo_url");
+        return _client.init(configEp(), "Persistence_Cluster_1")
+        .then([this] (){
+            return _client.create("LogStream1");
+        });
+    }
+
+    seastar::future<> runTest3() {
+        K2INFO(">>> Test3: Write and Read a Value from log stream");
+        String header;
+        for (uint32_t i = 0; i < 10000; ++i){
+            header = header + "0";
+        }
+        Payload payload([] { return Binary(4096); });
+        payload.write(header);
+        return _client.write(std::move(payload))
+        .then([this, header] (){
+            return _client.read("LogStream1");
+        })
+        .then([this, header] (auto&& payloads){
+            String str;
+            for (auto& payload:payloads){
+                payload.seek(0);
+                payload.read(str);
+                K2EXPECT(str, header);
+            }
+            return seastar::make_ready_future<>();
+        });
+    }
+
+    seastar::future<> runTest4() {
```

Comment:
How many wal plogs does this use? How many metadata plogs? Does it exhaust the wal plog pool?

Reply:
For the current test, it will use 3 WAL plogs. Now I set the size of WAL plog pool to be 1, and create new plogs if the current WAL plog pool is empty. But currently, this test just uses 1 metadata plog. It is difficult for me to test the scenario that we have used more than 1 metadata plogs. I reduce the size of each plog from 16M to 100Bytes by my own to test this scenario. But in our code, we did not support changing the size of each plog. So I cannot implement this test. Do you think we should add this to our code as well?

Reply:
Maybe you can add a very small plog size for testing purposes?

Reply:
In our code, we use `constexpr static uint32_t PLOG_MAX_SIZE = 16 * 1024 * 1024;` to set the size of each plog. Do you think we should change this to be a variable and allow our plog client to set this value in the plog server?

Reply:
I see, you can leave it how it is, it isn't a big deal.

Reply:
leave a TODO here